README.md

```diff
@@ -147,6 +148,43 @@ WASI host must:
 A WASI host that implements the above should be able to spawn threads for a
 variety of languages.
 
+### Thread group
+
+* The main thread starts with a thread group which only contains
+  the main thread.
+
+* Threads created by a thread in a thread group using `wasi_thread_spawn`
+  is added to the thread group.
+
+* When a thread is terminated, it's removed from the thread group.
+
+### Voluntary thread termination
+
+A thread can terminate itself voluntarily, either by calling
+`wasi_thread_exit`, or by returning from `wasi_thread_start`.
+
+### Changes to WASI `proc_exit`
+
+With this proposal, the `proc_exit` function takes extra responsibility
+to terminate all threads in the thread group, not only the calling one.
+
+Any of threads in the thread group can call `proc_exit`.
+
+### Traps
+
+When a thread caused a trap, it terminates all threads in the thread group
+similarly to `proc_exit`.
+
+### Thread group exit status
```

Comment:
How about `Process exit status`  ?   This terminology makes sense to me and aligns with the `proc_exit` name (the proc here stands for `process`)